reconcile/terraform_aws_route53.py

```diff
@@ -0,0 +1,137 @@
+
+QONTRACT_INTEGRATION = 'terraform_aws_route53'
+QONTRACT_INTEGRATION_VERSION = semver.format_version(0, 1, 0)
+
+
+def build_desired_state(zones):
+    """
+    Build the desired state from the app-interface resources
+
+    :param zones: List of zone resources to build state for
+    :type zones: list of dict
+    :return: State
+    :rtype: list of dict
+    """
+
+    desired_state = []
+    for zone in zones:
+        account = zone['account']
+        account_name = account['name']
+
+        zone_name = zone['name']
+        zone_values = {
+            'name': zone_name,
+            'account_name': account_name,
+            'records': []
+        }
+
+        # Check if we have unmanaged_record_names (urn) and compile them
+        # all as regular expressions
+        urn_compiled = []
+        for urn in zone.get('unmanaged_record_names', []):
+            urn_compiled.append(re.compile(urn))
+
+        for record in zone['records']:
+            record_name = record['name']
+
+            # Check if this record should be ignored
+            # as per 'unmanaged_record_names'
+            ignored = False
+            for regex in urn_compiled:
+                if regex.fullmatch(record['name']):
+                    logging.debug(f'{zone_name}: excluding unmanaged '
+                                  f'record {record_name} because it matched '
+                                  f'unmanaged_record_names pattern '
+                                  f'\'{regex.pattern}\'')
+                    ignored = True
+            if ignored:
+                continue
+
+            # We use the record object as-is from the list as the terraform
+            # data to apply. This makes things simpler and map 1-to-1 with
+            # Terraform's capabilities. As such we need to remove (pop) some of
+            # the keys we use for our own features
+
+            # Process '_target_cluster'
+            target_cluster = record.pop('_target_cluster', None)
+            if target_cluster:
+                target_cluster_elb = target_cluster['elbFQDN']
+                logging.debug(f'{zone_name}: field `_target_cluster` found '
+                              f'for record {record_name}. Value will be '
+                              f'{target_cluster_elb}')
+                record['records'] = [target_cluster_elb]
+
+            # Process '_healthcheck'
+            healthcheck = record.pop('_healthcheck', None)
+            if healthcheck:
+                logging.debug(f'{zone_name}: field `_healthcheck` found '
+                              f'for record {record_name}. Values are: '
+                              f'{healthcheck}')
+                record['healthcheck'] = healthcheck
+
+            zone_values['records'].append(record)
+
+        desired_state.append(zone_values)
+    return desired_state
+
+
+@defer
+def run(dry_run=False, print_only=False,
+        enable_deletion=False, thread_pool_size=10, defer=None):
```

Comment:
i assume we would want to change `enable_deletion` to true once we gain confidence?

Reply:
That was a (convenient?) oversight from my part. I think it's safe(r) to leave it disabled for the initial run and I'll enabled it shortly after